src/Raven.Server/Web/System/AdminServerWideHandler.cs

```diff
@@ -247,7 +255,10 @@ await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBody
                     result.Results.Add(configuration);
                 }
 
-                context.Write(writer, result.ToJson());
+                if(writeResponseBodyStream)
```

Comment:
let's avoid that

Better to split the method into 3:
1.
```
private async Task<ServerWideTasksResult<T>> GetTaskConfigurationsAsync<T>(OngoingTaskType type, Func<BlittableJsonReaderObject, T> converter)
{
// open context
// GetTaskConfigurationsInternalAsync
}
```
2.
```
private async Task WriteTaskConfigurationsAsync<T>(OngoingTaskType type, Func<BlittableJsonReaderObject, T> converter)
{
// open context
// GetTaskConfigurationsInternalAsync
// open writer and write
}
```
3.
```
private async Task<ServerWideTasksResult<T>> GetTaskConfigurationsInternalAsync<T>(TransactionOperationContext context, OngoingTaskType type, Func<BlittableJsonReaderObject, T> converter)
{
// open read tx
// prepare the data
}
```

Reply:
Done.